### PR TITLE
Mainnet migration, new GP configuration

### PR DESCRIPTION
**Pull request by dkent600**
This contains the new GP configuration parameters for the mainnet deploy.

### CODE REVIEW

**Comment by dkent600** [comment]
[Deploy contracts on the main net](https://trello.com/c/4aSAr2Fv/140-deploy-contracts-on-the-main-net)

File: test/genesisProtocol.ts [hunk 1]
```diff
@@ -170,8 +172,9 @@ describe("GenesisProtocol", () => {
       avatar: dao.avatar.address,
     });
     assert.isOk(result);
-    assert.equal(helpers.fromWei(result.thresholdConstA).toNumber(), 2);
-    assert.equal(result.thresholdConstB, 10);
+    assert.equal(helpers.fromWei(result.thresholdConstA).toNumber(),
```
Comment (dev-matan-tsuberi) [inline test/genesisProtocol.ts]: not strictly needed, but consider using https://www.npmjs.com/package/chai-bignumber for asserting on bignumbers.

File: lib/wrappers/genesisProtocol.ts [hunk 2]
```diff
@@ -1312,16 +1312,16 @@ export enum ProposalState {
 export const GetDefaultGenesisProtocolParameters = async (): Promise<GenesisProtocolParams> => {
   const web3 = await Utils.getWeb3();
   return {
-    boostedVotePeriodLimit: 604800, // 1 week
+    boostedVotePeriodLimit: 604800,
```
Comment (dev-matan-tsuberi) [inline lib/wrappers/genesisProtocol.ts]: should be 3 days (259200) instead of a week, no?

File: lib/migrations/2_deploy_organization.ts [hunk 3]
```diff
@@ -110,12 +110,25 @@ export const arcJsDeployer = (web3: Web3, artifacts: any, deployer: any): void =
...
+
+    if (network !== "live") {
+      // create a global GEN token
+      // await deployer.deploy(DaoToken, tokenName, tokenSymbol, 0);
+      // genTokenAddress = (await DaoToken.deployed()).address;
+      genTokenAddress = nativeTokenAddress;
+      console.log(`using global GEN token for network != "live" at: ${genTokenAddress}`);
+    } else {
+      // the "real" live ETH GEN token
+      genTokenAddress = "0x543Ff227F64Aa17eA132Bf9886cAb5DB55DCAddf";
```
Comment (tibetsprague) [inline lib/migrations/2_deploy_organization.ts]: how come this address isn't the Avatar's nativeToken()? shouldn't it be?

File: lib/wrappers/genesisProtocol.ts [hunk 4]
```diff
@@ -910,62 +910,62 @@ export interface RedeemEventResult {
...
-   * Default is 5, converted to Wei.
+   * Default is 50, converted to Wei.
    */
   proposingRepRewardConstA: BigNumber.BigNumber | string;
   /**
    * Constant B in the calculation of the proposer's reward.
    * See [[GenesisProtocolWrapper.getRedeemableReputationProposer]].
    * Must be between 0 and 100.
-   * Default is 1.
+   * Default is 50.
```
Comment (tibetsprague) [inline lib/wrappers/genesisProtocol.ts]: im a bit concerned that this is too big. remember https://github.com/daostack/arc.js/issues/196? arc may need to change to treat it as a percentage right? we will test though. @orenyodfat @fmatan

File: lib/migrations/2_deploy_organization.ts (hunk 3 above)
Comment (tibetsprague) [reply lib/migrations/2_deploy_organization.ts]: it is? how is it saying that? it seems to me that GEN should always be the native token of the Genesis DAO when deployed on any network

Comment (dev-matan-tsuberi) [reply lib/migrations/2_deploy_organization.ts]: this gets passed to the constructor of `GenesisProtocol` which accepts the staking token.

Comment (dev-matan-tsuberi) [reply lib/migrations/2_deploy_organization.ts]: https://github.com/daostack/arc/blob/master/contracts/VotingMachines/GenesisProtocol.sol#L96

**Pull request merged**